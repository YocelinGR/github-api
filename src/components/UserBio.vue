<template>
  <div class="row">
    <div class="col-md-3 user-space">
      <div class="card user" style="width: 18rem;">
        <img :src="user.avatarUrl" class="card-img-top" alt="...">
        <div class="card-body">
          <span class="emoji">{{ user.status && user.status.emoji}}</span>
          <p class="card-text user-message">{{ user.status && user.status.message}}</p>
        </div>
      </div>
      <div class="user-description">
        <p>{{ user.login}}</p>
        <p>{{ user.bio }}</p>
        <p><i class='fas fa-map-marker-alt' style='font-size:24px;color:gray'></i>
        {{  user.location }}</p>
        <button type="button" class="btn btn-secondary btn-block">Edit</button>
      </div>
      <hr>
    </div>
    <div class="col-md-8 user-work">
      <nav class="navbar navbar-expand-lg navbar-light bg-light">
        <div id="navbarNavAltMarkup" class="collapse navbar-collapse">
          <div class="navbar-nav">
            <a class="nav-item nav-link" href="#">Overview</a>
            <a class="nav-item nav-link" href="#">Repositories</a>
            <a class="nav-item nav-link" href="#">Projects</a>
            <a class="nav-item nav-link" href="#">Stars</a>
            <a class="nav-item nav-link" href="#">Followers {{ user.followers && user.followers.totalCount}}</a>
            <a class="nav-item nav-link" href="#">Following {{ user.followers && user.following.totalCount}}</a>
          </div>
        </div>
      </nav>
    </div>
  </div>
</template>

<script>
  import gql from 'graphql-tag';

    const getUser = gql`
      query user {
      user(login: "YocelinGR"){
        avatarUrl(size: 256),
        bio,
        status{
          emoji,
          message
        },
        company,
        createdAt,
        databaseId,
        id,
        location,
        login,
        followers {
          totalCount
        },
        following {
          totalCount
        },
      }
    }
    `;
    export default {
      data: () => ({
        user: {},
      }),
      apollo: {
        user: {
          query: getUser,
          }
        }
    }
</script>

<style>
.user-message{
  box-sizing: border-box;
  display: block;
  font-size: 20px;
  color: #24292e;
  white-space: normal;
}
.user {
  justify-content: center;
  text-align: center;
  vertical-align: center;
}
.emoji {
  position:relative;
  top: -32px;
  left: -60px;
  padding: 6px;
  background-color: rgb(224, 172, 60);
  border-radius: 15px;
}
.user-description{
  justify-content: center;
  padding-top: 15px;
  padding-left: 5px;
  font-size: 18px;
  color: rgb(92, 90, 90);
  text-align: justify;
  vertical-align: center;
}
.user-space {
  max-width: 100%;
  padding-top: 30px;
  padding-left: 30px;
}

.user-work {
  max-width: 100%;
  padding-top: 15px;
  padding-left: 5px;
}
</style>
