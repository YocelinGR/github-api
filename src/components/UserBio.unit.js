import UserBio from './UserBio.vue'
import { createLocalVue } from '@vue/test-utils'

describe('@components/UserBio', () => {
  const localVue = createLocalVue()
  it('UserBio is a vue instance', () => {
    const wrapper = shallowMount(UserBio, { localVue })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('Displays user starred repositories data', () => {
    const wrapper = shallowMount(UserBio, { localVue })
    wrapper.setData({
      user: {
         avatarUrl: "https://avatars0.githubusercontent.com/u/39833045?s=256&v=4",
         id: "MDQ6VXNlcjM5ODMzMDQ1",
         bio: "Front End Developer egresada de Laboratoria e\r\nIngeniera en Comunicaciones y Electrónica.\r\nApasionada por la programación y fiel creyente de\r\nla mejora continua",
         location: "Estado de México, CP 52928",
          status: {
            emoji: ":shipit:",
            message: "Studying "
          },
          followers: {
            totalCount: 18
          },
          following: {
            totalCount: 2
          },
          projects: {
            totalCount: 2
          },
          repositories: {
            totalCount: 42
          },
          starredRepositories: {
            totalCount: 2
          }
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
