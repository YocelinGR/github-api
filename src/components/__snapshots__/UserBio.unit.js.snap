// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/UserBio Displays user starred repositories data 1`] = `
<div
  class="row"
>
  <div
    class="col-md-3 user-space"
  >
    <div
      class="card user"
      style="width: 18rem;"
    >
      <img
        alt="..."
        class="card-img-top"
        src="https://avatars0.githubusercontent.com/u/39833045?s=256&v=4"
      />
       
      <div
        class="card-body"
      >
        <span
          class="emoji"
          data-markdown=""
        >
          
            :shipit:
        
        </span>
         
        <p
          class="card-text user-message"
        >
          Studying 
        </p>
      </div>
    </div>
     
    <div
      class="user-description"
    >
      <p>
        
      </p>
       
      <p>
        Front End Developer egresada de Laboratoria e
Ingeniera en Comunicaciones y Electrónica.
Apasionada por la programación y fiel creyente de
la mejora continua
      </p>
       
      <p>
        <i
          class="fas fa-map-marker-alt"
          style="font-size: 24px; color: gray;"
        />
        
      Estado de México, CP 52928
      </p>
       
      <button
        class="btn btn-secondary btn-block"
        type="button"
      >
        Edit
      </button>
    </div>
     
    <hr />
  </div>
   
  <div
    class="col-md-8 user-work"
  >
    <githubtab-stub
      totalcount="[object Object]"
    />
  </div>
</div>
`;
